utils/utils.go

```diff
@@ -25,6 +26,9 @@ func DownloadWithProgress(url string, tarName string, destFolder string) (err er
 	if err != nil {
 		return err
 	}
+	if resp.StatusCode == http.StatusNotFound {
+		return fmt.Errorf("%s 404 not found", url)
+	}
 
 	defer func(Body io.ReadCloser) {
```

Comment:
And I think we should check status code after this defer. Body should be closed if we return error code.

Reply:
Thanks